ICT4DNews/app/src/main/java/at/ict4d/ict4dnews/screens/news/detail/ICT4DNewsDetailActivity.kt

```diff
@@ -25,10 +28,12 @@ class ICT4DNewsDetailActivity : BaseActivity<ICT4DNewsDetailViewModel, ActivityI
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-
         fab.setOnClickListener { view ->
-            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
-                .setAction("Action", null).show()
+            val linkShare = intent.getParcelableExtra<News>(KEY_NEWS_LIST_MODEL)
+            val sharingIntent = Intent(Intent.ACTION_SEND)
+            sharingIntent.putExtra(Intent.EXTRA_TEXT, linkShare.link)
+            sharingIntent.setType("text/plain")
+            this.startActivity(Intent.createChooser(sharingIntent, resources.getText(R.string.send_to)))
```

Comment:
We use ANKO in our project which makes programming Android in Koltin much easier, you should check it out:
https://github.com/Kotlin/anko/wiki/Anko-Commons-%E2%80%93-Intents

The provide Intent Extensions you can replace you code simply with: `share(text, [subject])`, pretty cool 😉 
Please use it!

Please also replace the String with:

```
[Title of the Blog Post] by [Author] - [date]
[link to the blog post]```